### PR TITLE
Python scripts for build [ECR-4308]

### PR DESCRIPTION
**Pull request by vitvakatu**
## Overview

This is the experimental re-implementation of `package_app.sh` and `run_all_tests.sh` scripts in Python. Should work on Windows, Mac and Linux. It does not support some existing features, namely build settings (`--skip-tests`, `--release`, `--skip-cargo-clean`), but it can be easily added in the future.

__Please note: there are possible improvements to Python code to reduce code duplication, improve structure and readability, but this PR is for review of a general approach__

Pros:
- Python is more readable than bash or Powershell
- No code duplication between bash and powershell scripts

Cons:
- Triple code duplication: tests_profile.sh, tests_profile.ps1, tests_profile.py

---
See: https://jira.bf.local/browse/ECR-4308

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [ ] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [ ] Public API has Javadoc
- [ ] Method preconditions are checked and documented in the Javadoc of the method
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding/tests_profile.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        self.java_binding_lib_name = self._java_bindings_lib_name()
+        self.exonum_java_name = self._exonum_java_name()
+
+        self._set_environment_variables()
+
+    def _set_environment_variables(self):
+        os.environ["JAVA_HOME"] = self.java_home
+        print(f"JAVA_HOME={self.java_home}")
+        self._set_rustflags()
+        os.environ["PATH"] = self._create_path()
```
Comment (dmitry-timofeev) [inline exonum-java-binding/tests_profile.py]: We didn't use to modify `PATH` (on Linux/Mac). Why is that needed?

File: exonum-java-binding/tests_profile.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    def _create_path(self):
+        return os.environ["PATH"] + os.pathsep + \
+               self.java_lib_dir + os.pathsep + \
+               self.std_lib_dir + os.pathsep + \
+               os.path.join(self.project_root, "core", "rust", "target", "debug")
+
+    def _set_rustflags(self):
+        if sys.platform == 'win32':
+            new_rustflags = " "
```
Comment (dmitry-timofeev) [inline exonum-java-binding/tests_profile.py]: A space, not empty?

File: exonum-java-binding/package_app.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    os.environ["RUSTFLAGS"] = rustflags
+
+
+def clear_cargo(manifest_dir):
+    manifest_path = os.path.join(manifest_dir, "Cargo.toml")
+    subprocess.run(["cargo", "clean", "--manifest-path", manifest_path], check=True)
+
+
+if __name__ == '__main__':
+    tests_profile = TestsProfile()
```
Comment (dmitry-timofeev) [inline exonum-java-binding/package_app.py]: As tests profile sets different rust flags, overwritten here, shall it be created here?

Comment (vitvakatu) [reply exonum-java-binding/package_app.py]: Yes, probably we can implement `PackageProfile` subclass, I also noticed that.

File: exonum-java-binding/tests_profile.py (hunk 2 above)
Comment (vitvakatu) [reply exonum-java-binding/tests_profile.py]: 🤷‍♂ Copied from old windows scripts revision where wasn't such change

File: exonum-java-binding/tests_profile.py (hunk 1 above)
Comment (vitvakatu) [reply exonum-java-binding/tests_profile.py]: It's needed on Windows, but yes, it must be removed on Linux/Mac
